File: Python/July_problems/problem350_2nd_july_2024.py
from typing import List
from collections import Counter


def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
    m1 = Counter(nums1)
    result = []
    for num in nums2:
        if m1[num] > 0:
            result.append(num)
            m1[num] -= 1
    return result


def problem350():
    pass


if __name__ == '__main__':
    problem350()

File: CPP/July_Problems/problem350_2nd_july_2024.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

vector<int> intersect(vector<int> &nums1, vector<int> &nums2)
{
    unordered_map<int, int> m1;
    for (const auto &num : nums1)
    {
        m1[num] += 1;
    }
    vector<int> result;
    for (const auto &num : nums2)
    {
        if (m1[num] > 0)
        {
            result.push_back(num);
            m1[num] -= 1;
        }
    }
    return result;
}

int main()
{
    return 0;
}